### ISSUE TITLE
Missing query parameters on OpenPointInTimeRequest

### ISSUE THREAD

**Issue by TanveshT** [elastic/elasticsearch-java#102]
While creating an `OpenPointInTimeRequest` if suppose we have add one more parameter to the request like `ignore_unavailable`, the current functionality is absent in the sdk.

Is this functionality planned or already exists in the sdk?

Was going through the documentation for `ElasticsearchClient` and found out that we can set some `TransportOptions` while initialising the client, is this the appropriate place to set the `ignore_unavailable` value? if yes then the entire client will be configured with the same setting whereas it must be for a particular type of request.

**Comment by TanveshT** [comment]
@elasticmachine @swallez

**Comment by swallez** [comment]
The `ignore_unavailable` parameter (and a few others) are indeed missing from `OpenPointInTimeRequest`. We'll fix this in a future release.

In the meantime, transport options are indeed the way to go. A key point of `withTransportOptions` (as stated in the method's javadoc) is that it creates a new client. Clients are very lightweight, network connections are managed by the transport object.

The best way to create the new client with the additional parameter to keep any existing options is as follows:

```java
ElasticsearchClient ignoreUnavailableClient = client.withTransportOptions(client._transportOptions().with(
    opt -> opt.setParameter("ignore_unavailable", "true"))
);
```

I've changed the issue title to better reflect the actual issue.

**Comment by TanveshT** [comment]
Thank you @swallez for responding to the issue, I will surely try the approach suggested

**Comment by swallez** [comment]
This has been included in versions 7.17.3 and 8.2.0
